.github/workflows/unit-test.yml

```
@@ -97,61 +107,129 @@ jobs:
           ref: ${{ inputs.branch }}
           fetch-depth: ${{ inputs.checkout_fetch_depth }}
 
+      - name: Determine benchmark name
+        if: ${{ inputs.pipeline_mode != 'unit-tests' }}
+        id: benchmark_name
+        run: |
+          job_name="${{ inputs.job_name }}"
+          _benchmark_name=${job_name//[(,)]/""}
+          _benchmark_name=${_benchmark_name//[(" ")]/"-"}
+          echo "benchmark_name=$_benchmark_name" >> $GITHUB_OUTPUT
+
+      - name: Cache reference benchmarks
+        if: ${{ inputs.pipeline_mode != 'unit-tests' }}
+        id: benchmark-cache
+        uses: actions/cache@v3
+        with:
+          path: ${{ github.workspace}}/benchmark_reference
+          key: ${{ steps.benchmark_name.outputs.benchmark_name }}-reference
+
+      - name: Check if the shared part of the job will continue
+        id: continue
+        run: >-
+          echo "confirm=$( echo
+          ${{ (
+            (inputs.pipeline_mode == 'unit-tests')
+            || (inputs.pipeline_mode == 'benchmarks')
+            || ((inputs.pipeline_mode == 'reference-benchmarks') && (steps.benchmark-cache.outputs.cache-hit != 'true')) )
+          }}
+          )" >> $GITHUB_OUTPUT
```

I don't think the echo and bash sub-shell is necessary here

```suggestion
          echo "confirm=${{
            contains(fromJSON('["unit-tests", "benchmarks"]'), inputs.pipeline_mode ||
            (
              inputs.pipeline_mode == 'reference-benchmarks' && 
              steps.benchmark-cache.outputs.cache-hit != 'true'
            ) 
          }}" >> $GITHUB_OUTPUT
```